extensions/sql-migration/src/dialog/autoRefreshSettingsDialog/autoRefreshSettingsDialog.ts

```diff
@@ -0,0 +1,128 @@
+
+import * as azdata from 'azdata';
+import * as EventEmitter from 'events';
+import { SupportedAutoRefreshIntervals } from '../../api/utils';
+import * as loc from '../../constants/strings';
+
+export class AutoRefreshSettingsDialog {
+	private _dialogObject!: azdata.window.Dialog;
+	private _view!: azdata.ModelView;
+	private _selectionEvent: EventEmitter = new EventEmitter();
+	private _selectedInterval!: SupportedAutoRefreshIntervals;
+
+	constructor(presetValue: SupportedAutoRefreshIntervals) {
+		this._dialogObject = azdata.window.createModelViewDialog(
+			'',
+			'AutoRefreshSettingsDialog',
+			250,
+			'callout',
+			'left',
+			false,
+			true,
+			<azdata.window.IDialogProperties>{
+				height: 16,
+				width: 16,
+				xPos: 0,
+				yPos: 0
+			});
+		this._selectedInterval = presetValue;
+	}
+
+	async initialize(): Promise<AutoRefreshSetting> {
+		let tab = azdata.window.createTab('');
+		await tab.registerContent(async (view: azdata.ModelView) => {
+			this._view = view;
+			const selectRefreshLabel = view.modelBuilder.text().withProps({
+				value: loc.SELECT_THE_REFRESH_INTERVAL,
+				CSSStyles: {
+					'font-size': '13px',
+					'font-weight': 'bold',
+				}
+			}).component();
+
+			const buttonGroup = 'refreshButtonGroup';
+
+			const option0 = view.modelBuilder.radioButton().withProps({
```

Comment:
I would use an array of options and use map function to return an array of options instead of flat things out here :)